model/src/application/interaction/mod.rs

```diff
@@ -300,13 +299,39 @@ impl<'de> Visitor<'de> for InteractionVisitor {
                     user,
                 });
 
-                match kind {
-                    InteractionType::ApplicationCommand => Self::Value::ApplicationCommand(command),
-                    InteractionType::ApplicationCommandAutocomplete => {
-                        Self::Value::ApplicationCommandAutocomplete(command)
-                    }
-                    _ => unreachable!(),
-                }
+                Self::Value::ApplicationCommand(command)
+            }
+            InteractionType::ApplicationCommandAutocomplete => {
+                let channel_id = channel_id.ok_or_else(|| DeError::missing_field("channel_id"))?;
+                let data = data
+                    .ok_or_else(|| DeError::missing_field("data"))?
+                    .deserialize_into()
+                    .map_err(DeserializerError::into_error)?;
+
+                let guild_id = guild_id.unwrap_or_default();
+                let guild_locale = guild_locale.unwrap_or_default();
+                let locale = locale.ok_or_else(|| DeError::missing_field("locale"))?;
+                let member = member.unwrap_or_default();
+                let user = user.unwrap_or_default();
+
+                #[cfg(feature = "tracing")]
+                tracing::trace!(%channel_id, "handling application command");
```

Comment:
```suggestion
                tracing::trace!(%channel_id, "handling application command autocomplete");
```
prolly too much a nit. But maybe better to be exact